x/lightclient/keeper/grpc_query_block_header.go

```diff
@@ -1,35 +1,33 @@
 package keeper
 
 import (
-	"context"
-	"fmt"
-
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/query"
 	"github.com/zeta-chain/zetacore/pkg/proofs"
-	"github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/x/lightclient/types"
+	"golang.org/x/net/context"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// GetAllBlockHeaders queries all for block header
-func (k Keeper) GetAllBlockHeaders(c context.Context, req *types.QueryAllBlockHeaderRequest) (*types.QueryAllBlockHeaderResponse, error) {
+// BlockHeaderAll queries all block headers
+func (k Keeper) BlockHeaderAll(c context.Context, req *types.QueryAllBlockHeaderRequest) (*types.QueryAllBlockHeaderResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 	ctx := sdk.UnwrapSDKContext(c)
 	store := ctx.KVStore(k.storeKey)
 	blockHeaderStore := prefix.NewStore(store, types.KeyPrefix(types.BlockHeaderKey))
 
-	var blockHeaders []*proofs.BlockHeader
+	var blockHeaders []proofs.BlockHeader
```

Comment:
NIT : we can add a function GetBlockHeadersPaginated , it would be easier to unit test

Reply:
I wouldn't see it as necessary since the query is the only function using the pagination option and the function is just doing this so the query would just call this function but I think we can add better test for pagination for the query

Reply:
For the tests: 1ee85715